third_party/validator/sql_database_instance.go

```diff
@@ -0,0 +1,230 @@
+// ----------------------------------------------------------------------------
+//
+//     This file is copied here by Magic Modules. The code for building up a
+//     sql database instance object is copied from the manually implemented
+//     provider file:
+//     third_party/terraform/resources/resource_sql_database_instance.go.erb.go
+//
+// ----------------------------------------------------------------------------
+package google
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"google.golang.org/api/googleapi"
+	sqladmin "google.golang.org/api/sqladmin/v1beta4"
+)
+
+func GetSQLDatabaseInstanceCaiObject(d TerraformResourceData, config *Config) (Asset, error) {
+	if obj, err := GetSQLDatabaseInstanceApiObject(d, config); err == nil {
+		return Asset{
+			Name: fmt.Sprintf("//cloudsql.googleapis.com/%s", obj["selfLink"]),
```

Comment:
I'm not sure if self links are always stored as either relative paths or fully qualified paths.  Rather than blindly using `self_link` you should check out `google/self_link_helpers.go` and see if there is a conversion function (maybe like getRelativePath) that will be tolerant of either format then wrap it around your access of self_links.

Reply:
Sounds good, yeah selfLink is a known issue right now: https://github.com/GoogleCloudPlatform/terraform-google-conversion/issues/10